File: JULY-2025/15-07-2025/p3136.cpp
class Solution {
public:
    bool isValid(string word) {
        int n=word.size();
        if(n<3)return false;
        int vow=0,cons=0,digit=1;
        string vowel="aeiouAEIOU";
        for(char c:word){
             if(!isalnum(c))return false; // neither letter nor digit
             else if(vowel.find(c)!=string::npos)vow=1; //vowel
             else if(!isdigit(c)) cons=1; // consonant
        }
        return (vow==1 && cons==1);
        
    }
};

File: JULY-2025/15-07-2025/p3136.java
class Solution {
    public boolean isValid(String word) {
        int len = word.length(), vowel = 0, consonant = 0;

        if (len < 3)
            return false;

        for (int i = 0; i < len; i++) 
        {
            if (Character.isLetterOrDigit(word.charAt(i))) 
            {
                if (word.charAt(i) == 'A' || word.charAt(i) =='E' ||    word.charAt(i) == 'I' || word.charAt(i) == 'O' || word.charAt(i) == 'U' || word.charAt(i) == 'a' || word.charAt(i) == 'e' || word.charAt(i) == 'i' || word.charAt(i) == 'o' || word.charAt(i) == 'u')
                    vowel++;

                else if (word.charAt(i) < '0' || word.charAt(i) > '9')
                    consonant++;
            }

            else
                return false;
        }

        if (vowel != 0 && consonant != 0)
            return true;
        else
            return false;
    }
}